sql/binlog_crypt_data.cc

```diff
@@ -61,49 +62,66 @@ Binlog_crypt_data& Binlog_crypt_data::operator=(Binlog_crypt_data b)
   return *this;
 }
 
-bool Binlog_crypt_data::init(uint sch, uint kv, const uchar* nonce)
+bool Binlog_crypt_data::load_latest_binlog_key()
 {
-  scheme= sch;
-  key_version= kv;
-  free_key();
-  key_length= 16;
-
+  free_key(key, key_length);
 #ifdef MYSQL_SERVER
-  DBUG_ASSERT(nonce != NULL);
-  memcpy(this->nonce, nonce, BINLOG_NONCE_LENGTH);
-
-  boost::movelib::unique_ptr<char, void (*)(void*)> key_type(NULL, my_free);
-  char *key_type_raw = NULL;
-  size_t key_len;
+  char *system_key_type = NULL;
+  size_t system_key_len = 0;
+  uchar *system_key = NULL;
 
   DBUG_EXECUTE_IF("binlog_encryption_error_on_key_fetch",
                   { return true; } );
 
-  int fetch_result = my_key_fetch("percona_binlog", &key_type_raw, NULL,
-                                  reinterpret_cast<void**>(&key), &key_len);
-  key_type.reset(key_type_raw);
-  if (fetch_result || (key != NULL && key_len != 16))
-  {
-    free_key();
+  if (my_key_fetch("percona_binlog", &system_key_type, NULL,
```

Comment:
Consider putting the string literal in a constant (we might have discussed this in the binlog encryption PR but I don't recall what was the outcome)

Reply:
Done